### PR TITLE
fix #4925 feat(nimbus): display recommended signoffs on review page

### LINKED ISSUE #4925: (FE) Display "Recommended actions before launch" table
After EXP-1129 is complete, the FE should have easy access to data like `{qa_signoff: true, vp_signoff: false, legal_signoff: true}` to display green `Recommended:` text where the sign off recommendation is `true`.

Read this data and display the able according to Figma on the Request Review page.

See Figma:  https://www.figma.com/file/5SLa07XqQb0ttB1ay1gami/Nimbus-Console-UI?node-id=218778%3A23479

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-1132)
┆Issue Number: EXP-1132

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* After risk mitigation questions are filled out we need to show which sign offs are recommended

This commit

* Displays the recommended sign offs table on the review page

Screenshot 2021-05-05 at 12-39-28 Expanded didactic access – Review Launch Experimenter

Blocked on getting the three `Learn More` links from @AnaMedinac

### CODE REVIEW

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx [hunk 1]
```diff
@@ -161,6 +162,46 @@ const PageRequestReview = ({
               </ChangeApprovalOperations>
             )}
 
+            <h5>Recommended actions before launch</h5>
+            <Table bordered data-testid="table-signoff" className="mb-4">
+              <tr data-testid="table-signoff-qa">
+                <td>
+                  <strong>QA Sign - Off</strong>
```
Comment (jodyheavener) [inline app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:169]: Drive-by feedback: can we pretty please make "Sign - Off" one word, "signoff", or make it a hyphenated word, "sign-off"? The way it currently is makes it seem like an en-dash

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx [hunk 2]
```diff
@@ -161,6 +162,46 @@ const PageRequestReview = ({
               </ChangeApprovalOperations>
             )}
 
+            <h5>Recommended actions before launch</h5>
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:165]: This should actually be an `h3`. We probably need to tweak some of these since I see some are `h2`s when "Review and Launch" is an `h2`, could be a low priority maintenance item.

Also! We use `mb-3` on all other headers on this page.

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx [hunk 3]
```diff
@@ -161,6 +162,46 @@ const PageRequestReview = ({
...
+              <tr data-testid="table-signoff-qa">
+                <td>
+                  <strong>QA Sign - Off</strong>
+                </td>
+                <td>
+                  {experiment.signoffRecommendations?.qaSignoff && (
+                    <span className="text-success">Recommended: </span>
+                  )}
+                  Describe what they should do.
+                  <a href=""> Learn more</a>
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:176]: I know you'll use `LinkExternal` here when you've got these😎

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx [hunk 4]
```diff
@@ -161,6 +162,46 @@ const PageRequestReview = ({
               </ChangeApprovalOperations>
             )}
 
+            <h5>Recommended actions before launch</h5>
+            <Table bordered data-testid="table-signoff" className="mb-4">
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:166]: This is not entirely necessary but it might be nice to have this as a separate component, mostly because the Request Review page has a ton of stories already and I'd like to see the various states reflected there!

image

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx [hunk 5]
```diff
@@ -161,6 +162,46 @@ const PageRequestReview = ({
...
+              <tr data-testid="table-signoff-qa">
+                <td>
+                  <strong>QA Sign - Off</strong>
+                </td>
+                <td>
+                  {experiment.signoffRecommendations?.qaSignoff && (
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:172]: I think `qaSignoff` will always be `true` right? I guess it's good to leave the check in here anyway alongside the test, though if there was any possibility of this being `false` I can't help but think it'd be a little confusing without at least one `Recommended:` text guaranteed. As a user I'd just assume all of these were recommended🤷‍♀️

image

**Review by LZoog — COMMENTED** [review]
Looking gewd so far! Mostly nits, though it would be really nice to see Storybook states for a few different states (all, none (or only `qaSignoff` if that'll always be `true`), and maybe two recommendations just to cover some basis).

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx (hunk 2 above)
Comment (jaredlockhart) [reply app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:165]: That's not too big? In the mockups it looked like it should be smaller than the other titles?

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx (hunk 4 above)
Comment (jaredlockhart) [reply app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:166]: Yeah I wasn't sure if it was worth pulling into its own component

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx (hunk 5 above)
Comment (jaredlockhart) [reply app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:172]: Yeah but the assumption that it's always true is already hard coded in the backend, so if we hard code it here then it's done in two places.  And yeah I think what @AnaMedinac  was going for is that like, they're all kinda recommended, but if a risk mitigation question is true then they're a bit MORE recommended?  I think?

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx (hunk 2 above)
Comment (LZoog) [reply app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:165]: If the font needs to be smaller it can be done with classes, semantically speaking an `h5` should only occur if it's under an `h4` on the page!

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx (hunk 5 above)
Comment (LZoog) [reply app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:172]: Yeah I agree, just leave 'em in one place so if it ever changes we don't have to update the FE.

Also that sounds right as well, it's probably also fine since it says "recommended" above and not "required"

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx (hunk 4 above)
Comment (LZoog) [reply app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:166]: Actually maybe you can sneak it in like, "Preview status, one recommended sign off," since you already have "Review with recommended sign offs" which I somehow missed since the story list there is so long😂

**Comment by jodyheavener** [comment]
@jaredlockhart once you get the updated Learn More links you'll want to update the [constants](https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/src/lib/constants.ts#L29:L34) I introduced in #5170. Tbh I forgot about the outdated links before merging, so I'm glad your PR updates them 😆

**Comment by jaredlockhart** [comment]
Screenshot 2021-05-06 at 11-39-17 Reduced logistical system engine – Review Launch Experimenter

**Review by LZoog — APPROVED** [review]
LGTM besides those missing links, I'll go ahead and r+ so you can merge as soon as Ana has those to prevent more PR blockage 👍

File: app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx (hunk 1 above)
Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx:170]: @jaredlockhart 😢
